packages/deployments/contracts/contracts/core/connext/libraries/AssetLogic.sol

```diff
@@ -195,40 +225,53 @@ library AssetLogic {
     }
 
     // Swap the asset to the proper local asset
-    return _swapAsset(id, _asset, adopted, _amount);
+    return _swapAsset(SwapAssetArgs(id, _asset, adopted, _amount, _slippageTol));
   }
 
   /**
    * @notice Swaps assetIn t assetOut using the stored stable swap or internal swap pool
    * @dev Will not swap if the asset passed in is the adopted asset
-   * @param _canonicalId - The canonical token id
-   * @param _assetIn - The address of the from asset
-   * @param _assetOut - The address of the to asset
-   * @param _amount - The amount of the local asset to swap
+   * @param _args The SwapAssetArgs struct
    * @return The amount of assetOut
    * @return The address of assetOut
    */
-  function _swapAsset(
-    bytes32 _canonicalId,
-    address _assetIn,
-    address _assetOut,
-    uint256 _amount
-  ) internal returns (uint256, address) {
+  function _swapAsset(SwapAssetArgs memory _args) internal returns (uint256, address) {
     AppStorage storage s = LibConnextStorage.connextStorage();
 
     // Swap the asset to the proper local asset
+    uint8 tokenIndexIn;
+    uint8 tokenIndexOut;
 
-    if (stableSwapPoolExist(_canonicalId)) {
+    uint256 minReceived = (_args.amount * _args.slippageTol) / s.LIQUIDITY_FEE_DENOMINATOR;
+
+    if (stableSwapPoolExist(_args.canonicalId)) {
       // if internal swap pool exists
-      uint8 tokenIndexIn = getTokenIndexFromStableSwapPool(_canonicalId, _assetIn);
-      uint8 tokenIndexOut = getTokenIndexFromStableSwapPool(_canonicalId, _assetOut);
-      return (s.swapStorages[_canonicalId].swapInternal(tokenIndexIn, tokenIndexOut, _amount, 0), _assetOut);
+      tokenIndexIn = getTokenIndexFromStableSwapPool(_args.canonicalId, _args.assetIn);
+      tokenIndexOut = getTokenIndexFromStableSwapPool(_args.canonicalId, _args.assetOut);
+
+      // Enforce slippage
+      uint256 calculated = s.swapStorages[_args.canonicalId].calculateSwap(tokenIndexIn, tokenIndexOut, _args.amount);
+      if (calculated < minReceived) revert AssetLogic__swapAsset_highSlippage();
+
+      return (
+        s.swapStorages[_args.canonicalId].swapInternal(tokenIndexIn, tokenIndexOut, _args.amount, 0),
```

Comment:
```suggestion
        s.swapStorages[_args.canonicalId].swapInternal(tokenIndexIn, tokenIndexOut, _args.amount, minReceived),
```

Probably below code might be not necessary, because `swapInternal` function check that.
```
  // Enforce slippage
      uint256 calculated = s.swapStorages[_args.canonicalId].calculateSwap(tokenIndexIn, tokenIndexOut, _args.amount);
      if (calculated < minReceived) revert AssetLogic__swapAsset_highSlippage();
```

Reply:
Oh awesome, will make these changes!